File: abstractFactory/typescript/interfaces.ts
export interface GameParams {
  name: string
  type: string
}

export interface AppParams extends GameParams {
  price: number
}

export interface GameInstance {
  play: () => void
  getInfo(): GameParams
  printInformation: () => void
}

export interface AppInstance {
  start: () => void
  buyApp: (deposit: number) => boolean
  getInfo(): AppParams
  printInformation: () => void
}

export interface StoreFactory {
  createGame: (name: string) => GameInstance
  createApplication: (name: string) => AppInstance
}

export interface StoreInstance {
  run: (type: string, name: string) => void
  buyApp: (name: string, deposit: number) => void
  joinGame: (name: string) => void
  printInfo: (type: string, name: string) => void
}

File: Generating/abstractFactory/python/interfaces.py
from abc import ABC, abstractmethod

class GameInterface(ABC):
  @abstractmethod
  def play(self): pass
  @abstractmethod
  def getInfo(self): pass
  @abstractmethod
  def printInformation(): pass

class AppInterface(ABC):
  @abstractmethod
  def start(self): pass
  @abstractmethod
  def buyApp(self, deposit): pass
  @abstractmethod
  def getInfo(self): pass
  @abstractmethod
  def printInformation(self): pass

class StoreFactory(ABC):
  @abstractmethod
  def createGame(self, name): pass
  @abstractmethod
  def createApplication(self, name): pass

class StoreInstance(ABC):
  @abstractmethod
  def run(self, type, name): pass
  @abstractmethod
  def buyApp(self, name, deposit): pass
  @abstractmethod
  def joinGame(self, name): pass
  @abstractmethod
  def printInfo(self, type, name): pass